controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/CreateReaderGroupTask.java

```diff
@@ -57,48 +57,48 @@ public CreateReaderGroupTask(final StreamMetadataTasks streamMetaTasks,
-                               .thenCompose(configRecord -> {
-                               if (!ReaderGroupConfig.StreamDataRetention.values()[configRecord.getObject().getRetentionTypeOrdinal()]
-                                   .equals(ReaderGroupConfig.StreamDataRetention.NONE)) {
-                                   // update Stream metadata tables, only if RG is a Subscriber
-                                   Iterator<String> streamIter = configRecord.getObject().getStartingStreamCuts().keySet().iterator();
-                                   return Futures.loop(() -> streamIter.hasNext(), () -> {
-                                          Stream stream = Stream.of(streamIter.next());
-                                          return streamMetadataStore.addSubscriber(stream.getScope(),
-                                                       stream.getStreamName(), scopedRGName, configRecord.getObject().getGeneration(),
-                                                  null, executor);
-                                      }, executor);
-                                   }
-                               return CompletableFuture.completedFuture(null);
-                               }).thenCompose(v ->
-                                  streamMetadataTasks.createRGStream(scope, NameUtils.getStreamForReaderGroup(readerGroup),
-                                           StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(1)).build(),
-                                          System.currentTimeMillis(), 10)
-                                 .thenCompose(status -> {
-                                     if (status.equals(Controller.CreateStreamStatus.Status.STREAM_EXISTS)
-                                     || status.equals(Controller.CreateStreamStatus.Status.SUCCESS)) {
-                                         return Futures.toVoid(streamMetadataStore.updateReaderGroupVersionedState(scope, readerGroup,
-                                                 ReaderGroupState.ACTIVE, state, context, executor));
-                                     }
-                               return Futures.failedFuture(new IllegalStateException(String.format("Error creating StateSynchronizer Stream for Reader Group %s: %s",
-                                             readerGroup, status.toString())));
-                           })).exceptionally(ex -> {
-                                    log.debug(ex.getMessage());
-                                    Throwable cause = Exceptions.unwrap(ex);
-                                    throw new CompletionException(cause);
-                           });
+        return RetryHelper.withRetriesAsync(() -> streamMetadataStore.getReaderGroupId(scope, readerGroup, context, executor)
+                .thenCompose(rgId -> {
+                    if (!rgId.equals(readerGroupId)) {
+                        log.warn("Skipping processing of CreateReaderGroupEvent with stale UUID.");
```

Comment:
also this doesnt need to be warn, does it? its idempotent no op case. so we should add it at info and not warn

Reply:
Its an unexpected thing to happen hence added it as warn